gapic/templates/%namespace/%name_%version/%sub/services/%service/transports/base.py.j2

```diff
@@ -80,6 +97,26 @@ class {{ service.name }}Transport(abc.ABC):
             host += ':443'
         self._host = host
 
+        # If a custom API endpoint is set, set scopes to ensure the auth
+        # library does not used the self-signed JWT flow for service
+        # accounts
+        if (
+            host is not None
+            and self.DEFAULT_HOST is not None
```

Comment:
if `self.DEFAULT_HOST` is `None`, then shouldn't default scopes be used? Without a default host I do not think the library can self-sign a JWT.

Reply:
I thought about this a bit more, and I think this case (no default host) is taken care of in the auth library: https://github.com/googleapis/google-auth-library-python/blob/d80c85f285ae1a44ddc5a5d94a66e065a79f6d19/google/auth/transport/grpc.py#L82-L85.

The `if` statement got a bit convoluted because I was trying to avoid creating multiple branches (which would drop coverage). I changed `self.DEFAULT_HOST` to be a `str`, so I don't have to check `is not None` before parsing the string.